core/domain/translation_domain.py

```diff
@@ -381,3 +753,502 @@ def to_dict(self) -> Dict[str, str]:
             'source_text': self.source_text,
             'translated_text': self.translated_text
         }
+
+
+# WrittenTrasnlation and WrittenTranslations class is still used in topic and
+# subtopic entity and will be removed from here as well once Topic and subtopic
+# will also support translation feature.
+
+
+class WrittenTranslationDict(TypedDict):
+    """Dictionary representing the WrittenTranslation object."""
+
+    data_format: str
+    translation: Union[str, List[str]]
+    needs_update: bool
+
+
+class WrittenTranslation:
+    """Value object representing a written translation for a content.
+
+    Here, "content" could mean a string or a list of strings. The latter arises,
+    for example, in the case where we are checking for equality of a learner's
+    answer against a given set of strings. In such cases, the number of strings
+    in the translation of the original object may not be the same as the number
+    of strings in the original object.
+    """
+
+    DATA_FORMAT_HTML: Final = 'html'
+    DATA_FORMAT_UNICODE_STRING: Final = 'unicode'
+    DATA_FORMAT_SET_OF_NORMALIZED_STRING: Final = 'set_of_normalized_string'
+    DATA_FORMAT_SET_OF_UNICODE_STRING: Final = 'set_of_unicode_string'
+
+    DATA_FORMAT_TO_TRANSLATABLE_OBJ_TYPE: Dict[
+        str, translatable_object_registry.TranslatableObjectNames
+    ] = {
+        DATA_FORMAT_HTML: 'TranslatableHtml',
+        DATA_FORMAT_UNICODE_STRING: 'TranslatableUnicodeString',
+        DATA_FORMAT_SET_OF_NORMALIZED_STRING: (
+            'TranslatableSetOfNormalizedString'),
+        DATA_FORMAT_SET_OF_UNICODE_STRING: 'TranslatableSetOfUnicodeString',
+    }
+
+    @classmethod
+    def is_data_format_list(cls, data_format: str) -> bool:
+        """Checks whether the content of translation with given format is of
+        a list type.
+
+        Args:
+            data_format: str. The format of the translation.
+
+        Returns:
+            bool. Whether the content of translation is a list.
+        """
+        return data_format in (
+            cls.DATA_FORMAT_SET_OF_NORMALIZED_STRING,
+            cls.DATA_FORMAT_SET_OF_UNICODE_STRING
+        )
+
+    def __init__(
+        self,
+        data_format: str,
+        translation: Union[str, List[str]],
+        needs_update: bool
+    ) -> None:
```

Comment:
```suggestion
    def __init__(
        self,
        data_format: str,
        translation: Union[str, List[str]],
        *,
        needs_update: bool
    ) -> None:
```

that way we force the `needs_updated` to always be called with keyword.

Reply:
But why do we need needs_updated to be a keyword arg?

Reply:
To improve readability at the callsite.

Compare `save_something(exp_id, True, False)` and `save_something(exp_id, cache=True, override_previous_instance=False)`, I think the second one provides better understanding directly at the callsite and you only need to actually go inside the function if you are not sure what exactly these do, but in the first case you most probably need to always check the implementation to even see what the two bools do.

Reply:
> To improve readability at the callsite.

I see, I don't think we follow this pattern in our codebase, generally, I see we in this codebase) use keyword args where we have optional arguments or too many arguments. Also, in this case, it's just 3 arguments with just one boolean value, do you think it's necessary? (I'm fine doing it if you want :) )